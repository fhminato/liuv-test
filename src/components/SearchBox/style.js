import styled from 'styled-components/native';

export const SearchContainer = styled.View`
  display: flex;
  flex-direction: row;
  align-items: center;
  margin-bottom: 10px;
`;

export const InputContainer = styled.View`
  display: flex;
  width: 80%;
  justify-content: center;
`;

export const Button = styled.TouchableOpacity`
  justify-content: center;
  align-items: center;
  border-radius: 100px;
  margin: 15px 0px 0px 5px;
`;

export const ButtonText = styled.Text`
  color: #fff;
  text-transform: uppercase;
  font-family: 'Antipasto Pro Demibold';
`;

export const Text = styled.Text`
  color: #000;
  text-transform: uppercase;
  font-family: 'Antipasto Pro';
  margin-bottom: 5px;
`;

export const Input = styled.TextInput`
  height: 40px;
  width: 100%;
  border: 1px solid gray;
  border-radius: 5px;
  font-family: 'Antipasto Pro';
  font-size: 15px;
`;
