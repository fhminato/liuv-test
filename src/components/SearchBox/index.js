import React, {useState} from 'react';
import {Button, InputContainer, Input, Text, SearchContainer} from './style';
import {Image} from 'react-native';
import SearchIcon from '../../../assets/images/search-icon.png';

export default function SearchBox(props) {
  const [username, setUsername] = useState('');

  return (
    <SearchContainer>
      <InputContainer>
        <Text>Usuario</Text>
        <Input
          value={username}
          onChangeText={(username) => setUsername(username)}
        />
      </InputContainer>
      <Button onPress={() => props.onClick(username)}>
        <Image source={SearchIcon} style={{height: 30, width: 30}}></Image>
      </Button>
    </SearchContainer>
  );
}
