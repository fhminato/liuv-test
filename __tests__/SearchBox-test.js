import React from 'react';
import SearchBox from '../src/components/SearchBox';

import renderer from 'react-test-renderer';
import {Text} from 'react-native';

test('renders correctly', () => {
  const tree = renderer
    .create(<SearchBox onClick={() => console.log()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
