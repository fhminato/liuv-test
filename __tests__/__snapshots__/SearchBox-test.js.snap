// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "justifyContent": "center",
          "width": "80%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Antipasto Pro",
            "marginBottom": 5,
            "textTransform": "uppercase",
          },
        ]
      }
    >
      Usuario
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "gray",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 1,
            "fontFamily": "Antipasto Pro",
            "fontSize": 15,
            "height": 40,
            "width": "100%",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 100,
        "justifyContent": "center",
        "marginBottom": 0,
        "marginLeft": 5,
        "marginRight": 0,
        "marginTop": 15,
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/search-icon.png",
        }
      }
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
  </View>
</View>
`;

exports[`test text 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "justifyContent": "center",
          "width": "80%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Antipasto Pro",
            "marginBottom": 5,
            "textTransform": "uppercase",
          },
        ]
      }
    >
      Usuario
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "gray",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 1,
            "fontFamily": "Antipasto Pro",
            "fontSize": 15,
            "height": 40,
            "width": "100%",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 100,
        "justifyContent": "center",
        "marginBottom": 0,
        "marginLeft": 5,
        "marginRight": 0,
        "marginTop": 15,
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/search-icon.png",
        }
      }
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
  </View>
</View>
`;
